### PR TITLE
[IMP] account, l10n_latam_invoice_document, l10n_ar, l10n_cl: unify invoice report templates, add logic for custom reports, define l10n_ar nad l10n_cl custom reports 

### PR DESCRIPTION
**Pull request by zaoral**
task 306
---

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by ren-odoo** [comment]
Hi @jco-odoo 
I did the functional testing in the dummy enterprise version and it's working correctly, it's ready for your technical review. 

cc.  @zaoral @jjscarafia

File: addons/account/views/report_invoice.xml [hunk 1]
```diff
@@ -225,7 +222,8 @@
             <t t-call="web.html_container">
                 <t t-foreach="docs" t-as="o">
                     <t t-set="lang" t-value="o.invoice_user_id.sudo().lang if o.move_type in ('in_invoice', 'in_refund') else o.partner_id.lang"/>
-                    <t t-call="account.report_invoice_document_with_payments" t-lang="lang"/>
+                    <t t-set="print_with_payments" t-value="True"/>
```
Comment (jco-odoo) [inline addons/account/views/report_invoice.xml:225]: We could also integrate the get_custom_report function here and by default it returns the account.report_invoice_document

**Comment by jco-odoo** [comment]
@qdp-odoo Could you agree on this one for master?  The advantage is that if we want to do a primary view in the localizations (as some countries are really nasty in how the format of their invoices should look like) we do not have to inherit twice.  Right now, the invoices with payments is a primary view with no added value, if we can just pass it as a variable.

File: addons/account/views/report_invoice.xml (hunk 1 above)
Comment (zaoral) [reply addons/account/views/report_invoice.xml:225]: @jco-odoo  hi! ok, I like the idea. I made the push with the requested changes and leave it in a separate commit just for the review. Please let us know what do you think.

File: addons/account/views/report_invoice.xml [hunk 2]
```diff
@@ -225,7 +222,8 @@
...
+                    <t t-call="#{ o._get_custom_report('account.report_invoice_document') }" t-lang="lang"/>
```
Comment (jco-odoo) [inline addons/account/views/report_invoice.xml:226]: @zaoral We prefer not to pass the name of the report here, but _get_invoice_report(show_payments=False) as header instead.

Comment (zaoral) [reply addons/account/views/report_invoice.xml]: Hi @jco-odoo . Great, I removed the name of the report as parameter and return the name as the default one inside the method _get_custom_report(). You can see the change in this commit https://github.com/odoo/odoo/pull/54193/commits/94eccccc1ea8e331e19d5c39708da252ebdba82c

However I want to ask you, I did not see why you want to add show_payments=False in the method? Please let me know

File: addons/account/views/report_invoice.xml (hunk 1 above)
Comment (jco-odoo) [reply addons/account/views/report_invoice.xml:225]: @zaoral It should be named_get_invoice_report and indeed, the param is not necessary as we can set it through the t-set

Comment (zaoral) [reply addons/account/views/report_invoice.xml:225]: @jco-odoo it is ok if instead named_get_invoice_report the new name be something like _get_name_invoice_report?

**Comment by jco-odoo** [comment]
@zaoral I would like to have the fw-port of the other here as well  (separate commit than the changes in account)

**Comment by zaoral** [comment]
HI @jco-odoo 
This PR is ready for review.
It is related also to https://github.com/odoo/enterprise/pull/11717 please review too.

Best Regards

**Comment by jco-odoo** [comment]
@zaoral Can you also change the ADD to IMP and do a rebase (Certain commits might have been fw-ported in the meantime)

**Comment by zaoral** [comment]
HI @jco-odoo 
I have changed the ADD to IMP and do the rebase  for both this PR and https://github.com/odoo/enterprise/pull/11717
Is ready for your review

Best Regards

**Comment by jco-odoo** [comment]
@robodoo r+ rebase-ff

**Comment by jco-odoo** [comment]
@robodoo r+ rebase-ff

**Comment by nim-odoo** [comment]
This completely broke the edition of the invoice reports with Studio...

**Pull request closed without merging**